editors/vscode/src/client.ts

```diff
@@ -400,41 +400,53 @@ function getGoalsEnvContent(goals : Goal[]){
     let codeHyps : String = ""; //hypothesis HTML code
     let codeGoals : String = ""; //goals HTML code
     let codeEnvGoals : String = ""; //result code HTML
-
+    let goalNo = 0;
+    
     for(let i=0; i < goals.length; i++) {
+        
+        // check if this is a Type Goal
+        if ("gid" in goals[i]){
```

Comment:
This test looks to much like a hack. You seem to use the fact that in UnifGoal there is no "gid" field, but this could change in the future. It would be better to use a more robust test.

Reply:
Wouldn't typeofgoal be useful?

Reply:
I agree, but "Typ " had an extra space. I thought it was a bug. Anyway I will make the changes.